flytekit-java/src/main/java/org/flyte/flytekit/SdkWorkflow.java

```diff
@@ -85,10 +100,21 @@ public SdkType<OutputT> getOutputType() {
     return outputType;
   }
 
-  public WorkflowTemplate toIdlTemplate() {
+  public WorkflowTemplate expandToIdlTemplate() {
     SdkWorkflowBuilder builder = new SdkWorkflowBuilder();
-    this.expand(builder);
+    this.outerExpand(builder);
+    return toIdlTemplate(builder);
+  }
+
+  public WorkflowTemplate toIdlTemplate(SdkWorkflowBuilder builder) {
+    return WorkflowTemplateIdl.ofBuilder(builder, getSdkBindingDataInputs(), sdkBindingDataOutputs);
```

Comment:
```suggestion
    return WorkflowTemplateIdl.ofBuilder(builder, sdkBindingDataInputs, sdkBindingDataOutputs);
```
If we keep this method, then we can use the field directly